[Flaky unit test] pkg/kubelet/cri/remote TestVersion

**Which jobs are flaking**:
periodics-kubernetes-bazel-test

**Which test(s) are flaking**:
//pkg/kubelet/cri/remote   TestVersion
**Testgrid link**:

**Reason for failure**:
   Error:          Received unexpected error:
                                rpc error: code = Unavailable desc = transport is closing
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference

**Anything else we need to know**:
**To reproduce with stress:**
go test -race -c ./pkg/kubelet/cri/remote
stress ./remote.test -test.run TestVersion

Detailed logs
~~~
cloudusr@x86-arch-ra:~/kubernetes$ go test -race -c ./pkg/kubelet/cri/remote
cloudusr@x86-arch-ra:~/kubernetes$ stress ./remote.test -test.run TestVersion
/tmp/go-stress-20201130T052353-617378882
E1130 05:23:57.619773   11458 remote_runtime.go:86] Version from runtime service failed: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial unix /tmp/kubelet_remote.sock: connect: no such file or directory"
--- FAIL: TestVersion (0.00s)
    remote_runtime_test.go:60:
                Error Trace:    remote_runtime_test.go:60
                Error:          Received unexpected error:
                                rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial unix /tmp/kubelet_remote.sock: connect: no such file or directory"
                Test:           TestVersion
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd4c04d]

goroutine 7 [running]:
testing.tRunner.func1.1(0xdd2b80, 0x1365eb0)
        /usr/local/go/src/testing/testing.go:1072 +0x46a
testing.tRunner.func1(0xc0001ee000)
        /usr/local/go/src/testing/testing.go:1075 +0x636
panic(0xdd2b80, 0x1365eb0)
        /usr/local/go/src/runtime/panic.go:975 +0x47a
k8s.io/kubernetes/pkg/kubelet/cri/remote.TestVersion(0xc0001ee000)
        /home/cloudusr/kubernetes/pkg/kubelet/cri/remote/remote_runtime_test.go:61 +0x1ad
testing.tRunner(0xc0001ee000, 0xebcdc8)
        /usr/local/go/src/testing/testing.go:1123 +0x203
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1168 +0x5bc


ERROR: exit status 2
~~~

--- a/pkg/kubelet/cri/remote/BUILD
+++ b/pkg/kubelet/cri/remote/BUILD
@@ -52,6 +52,7 @@ go_test(
     embed = [":go_default_library"],
     deps = [
         "//pkg/kubelet/cri/remote/fake:go_default_library",
+        "//pkg/kubelet/cri/remote/util:go_default_library",
         "//staging/src/k8s.io/cri-api/pkg/apis:go_default_library",
         "//staging/src/k8s.io/cri-api/pkg/apis/testing:go_default_library",
         "//vendor/github.com/stretchr/testify/assert:go_default_library",

--- a/pkg/kubelet/cri/remote/fake/BUILD
+++ b/pkg/kubelet/cri/remote/fake/BUILD
@@ -23,7 +23,51 @@ go_library(
         "//staging/src/k8s.io/cri-api/pkg/apis/testing:go_default_library",
         "//vendor/google.golang.org/grpc:go_default_library",
         "//vendor/k8s.io/utils/exec:go_default_library",
-    ],
+    ] + select({
+        "@io_bazel_rules_go//go/platform:aix": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:android": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:darwin": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:dragonfly": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:freebsd": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:illumos": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:ios": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:js": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:linux": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:nacl": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:netbsd": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:openbsd": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:plan9": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "@io_bazel_rules_go//go/platform:solaris": [
+            "//staging/src/k8s.io/apimachinery/pkg/util/rand:go_default_library",
+        ],
+        "//conditions:default": [],
+    }),
 )
 
 filegroup(

--- a/pkg/kubelet/cri/remote/fake/endpoint.go
+++ b/pkg/kubelet/cri/remote/fake/endpoint.go
@@ -18,11 +18,17 @@ limitations under the License.
 
 package fake
 
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/rand"
+)
+
 const (
-	defaultUnixEndpoint = "unix:///tmp/kubelet_remote.sock"
+	defaultUnixEndpoint = "unix:///tmp/kubelet_remote_%v.sock"
 )
 
 // GenerateEndpoint generates a new unix socket server of grpc server.
 func GenerateEndpoint() (string, error) {
-	return defaultUnixEndpoint, nil
+	// use random int be a part fo file name
+	return fmt.Sprintf(defaultUnixEndpoint, rand.Int()), nil
 }

--- a/pkg/kubelet/cri/remote/remote_runtime_test.go
+++ b/pkg/kubelet/cri/remote/remote_runtime_test.go
@@ -17,6 +17,7 @@ limitations under the License.
 package remote
 
 import (
+	"os"
 	"testing"
 	"time"
 
@@ -25,6 +26,7 @@ import (
 	internalapi "k8s.io/cri-api/pkg/apis"
 	apitest "k8s.io/cri-api/pkg/apis/testing"
 	fakeremote "k8s.io/kubernetes/pkg/kubelet/cri/remote/fake"
+	"k8s.io/kubernetes/pkg/kubelet/cri/remote/util"
 )
 
 const (
@@ -53,7 +55,15 @@ func createRemoteRuntimeService(endpoint string, t *testing.T) internalapi.Runti
 
 func TestVersion(t *testing.T) {
 	fakeRuntime, endpoint := createAndStartFakeRemoteRuntime(t)
-	defer fakeRuntime.Stop()
+	defer func() {
+		fakeRuntime.Stop()
+		// clear endpoint file
+		if addr, _, err := util.GetAddressAndDialer(endpoint); err == nil {
+			if _, err := os.Stat(addr); err == nil {
+				os.Remove(addr)
+			}
+		}
+	}()
 
 	r := createRemoteRuntimeService(endpoint, t)
 	version, err := r.Version(apitest.FakeVersion)

